### PR TITLE
[API Gateway] Support access key authentication

### PR DESCRIPTION
**Pull request by TomerShor**
Add support for access key authentication mode, that uses the Cookie header with Iguazio access key for authentication.

Implements https://iguazio.atlassian.net/browse/ML-6359

### CODE REVIEW

**Review by liranbg — CHANGES_REQUESTED** [review]
condition that on minimum iguazio version (where you have the fix to support control planes

**Review by liranbg — APPROVED** [review]

File: mlrun/platforms/iguazio.py [hunk 1]
```diff
@@ -582,3 +582,48 @@ def sanitize_username(username: str):
...
+
+def min_iguazio_versions(*versions):
+    def decorator(function):
+        def wrapper(*args, **kwargs):
+            if validate_iguazio_version_compatibility(*versions):
+                return function(*args, **kwargs)
+
+            message = (
+                f"{function.__name__} is supported since Iguazio {' or '.join(versions)}, currently using "
+                f"Iguazio {mlconf.nuclio_version}."
```
Comment (alonmr) [inline mlrun/platforms/iguazio.py]: you are using `mlconf.nuclio_version` in the message

File: mlrun/runtimes/nuclio/api_gateway.py [hunk 2]
```diff
@@ -17,18 +17,21 @@
...
 
 import requests
+from nuclio.auth import AuthInfo as NuclioAuthInfo
 from requests.auth import HTTPBasicAuth
 
 import mlrun
 import mlrun.common.schemas
 
 from ...model import ModelObj
+from ...platforms.iguazio import min_iguazio_versions
```
Comment (alonmr) [inline mlrun/runtimes/nuclio/api_gateway.py]: dont add relative imports

File: mlrun/platforms/iguazio.py [hunk 3]
```diff
@@ -582,3 +582,48 @@ def sanitize_username(username: str):
...
+            )
+            raise mlrun.errors.MLRunIncompatibleVersionError(message)
+
+        return wrapper
+
+    return decorator
+
+
+def validate_iguazio_version_compatibility(*min_versions):
```
Comment (alonmr) [inline mlrun/platforms/iguazio.py]: I feel like this is code duplication maybe we can merge it with `validate_nuclio_version_compatibility`

File: tests/platforms/test_iguazio.py [hunk 4]
```diff
@@ -227,3 +229,23 @@ def test_is_iguazio_session_cookie():
...
+    mlconf.igz_version = "3.5.4"
+
+    for case in [
+        ["3.5.5"],
+        ["3.6.0"],
+        ["3.5.5-b25.20231224135202"],
+        ["4.0.0"],
+        ["3.2.0", "3.6.0"],
+    ]:
+        print(f"Testing case: {case}")
```
Comment (alonmr) [inline tests/platforms/test_iguazio.py]: change print to log and use pytest parameterize because it will continue for the other cases if one of them fails

**Review by alonmr — CHANGES_REQUESTED** [review]
Looks good! A few small comments

**Review by alonmr — APPROVED** [review]

**Pull request merged**